features/FEATURE_LWIP/lwip-interface/lwip-sys/arch/lwip_sys_arch.c

```diff
@@ -469,16 +517,16 @@ sys_thread_t sys_thread_new(const char *pcName,
     sys_thread_t t = (sys_thread_t)&thread_pool[thread_pool_index];
     thread_pool_index++;
     
-#ifdef CMSIS_OS_RTX
-    t->def.pthread = (os_pthread)thread;
-    t->def.tpriority = (osPriority)priority;
-    t->def.stacksize = stacksize;
-    t->def.stack_pointer = (uint32_t*)malloc(stacksize);
-    if (t->def.stack_pointer == NULL) {
```

Comment:
Some thread attributes aren't initialized here. The whole thread object should probably be memset to 0.